ambry-network/src/main/java/com/github/ambry/network/http2/AmbryNetworkRequestHandler.java

```diff
@@ -42,12 +48,19 @@ public AmbryNetworkRequestHandler(RequestResponseChannel requestResponseChannel,
   @Override
   protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest msg) {
     ByteBuf dup = msg.content().retainedDuplicate();
+    NettyServerRequest networkRequest = new NettyServerRequest(ctx, dup);
     try {
-      requestResponseChannel.sendRequest(new NettyServerRequest(ctx, dup));
+      discardHeaderBytes(networkRequest);
```

Comment:
Since we call discardHeaderBytes, suppose we'll remove the original place we remove the header.
But I didn't find the code in the PR.